### PR TITLE
Support custom SMTP properties for Email Connector

### LINKED ISSUE #22948: Support Custom SMTP properties with SMTP Client
**Description:**
Enable adding SMTP properties via SMTP Client

### PR DESCRIPTION
**Pull request by Maninda**
## Purpose
Add custom SMTP Properties to SMTP Client via its config.

Fixes #22948

## Approach
Add each of the property to `properties` map.

## Samples
A sample is give below where a single property, `mail.smtp.port` is added to the config, overriding the default properties.

```ballerina
email:SmtpConfig smtpConfig = {
    port: 30250, // This is an incorrect value. Later the correct value, 3025 will be set via a property.
    enableSsl: false,
    properties: {"mail.smtp.port":"3025"}
};
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/email/src/main/ballerina/src/email/smtp_client_endpoint.bal [hunk 1]
```diff
@@ -59,8 +59,10 @@ function send(SmtpClient clientEndpoint, Email email) returns Error? = @java:Met
 #
 # + port - Port number of the SMTP server
 # + enableSsl - If set to true, use SSL to connect and use the SSL port by default.
-#               The default value is true for the "smtps" protocol and false for the "smtp" protocol.
+#               The default value is true for the "smtps" protocol and false for the "smtp" protocol
+# + properties - SMTP properties to override the existing configuration
 public type SmtpConfig record {|
     int port = 465;
     boolean enableSsl = true;
+    Property[]? properties = ();
```
Comment (daneshk) [inline stdlib/email/src/main/ballerina/src/email/smtp_client_endpoint.bal:67]: Is it possible to make it as properties map like
map<string>? properties = ();

As we are converting the value to a string, I think we can restrict to `string` only
WDYT?

File: stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java [hunk 2]
```diff
@@ -74,6 +75,15 @@ public static Properties getProperties(MapValue smtpConfig, String host) {
...
         properties.put(EmailConstants.PROPS_ENABLE_SSL, smtpConfig.getBooleanValue(EmailConstants.PROPS_SSL));
+        CommonUtil.addCustomProperties(smtpConfig.getArrayValue(EmailConstants.PROPS_PROPERTIES), properties);
+        if (log.isDebugEnabled()) {
+            Set<String> propertySet = properties.stringPropertyNames();
+            log.debug("SMTP Properties set are as follows.");
+            for (Object propertyObj : propertySet) {
+                log.debug("Property Name: " + propertyObj + ", Value: " + properties.get(propertyObj).toString()
+                        + " ValueType: " + properties.get(propertyObj).getClass().getName());
+            }
+        }
```
Comment (daneshk) [inline stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java:86]: I think we need to give priority to the predefined fields? shall we first add custom properties and then add field values.

**Comment by riyafa** [comment]
Shall we have a proper title for the PR?

File: stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java (hunk 2 above)
Comment (Maninda) [reply stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java:86]: Custom properties are given higher priority to override the default functionality of the connector for advanced users.

**Comment by Maninda** [comment]
> Shall we have a proper title for the PR? It is not normal to have issue on the title as it doesn't mention the purpose of the PR

Corrected the title

File: stdlib/email/src/main/ballerina/src/email/smtp_client_endpoint.bal (hunk 1 above)
Comment (Maninda) [reply stdlib/email/src/main/ballerina/src/email/smtp_client_endpoint.bal]: Changed.

**Pull request merged**